cmake/modules/SwiftUtils.cmake

```diff
@@ -1,136 +1,174 @@
 
-  if (PRECONDITION_NEGATE)
-    if (${var})
-      if (PRECONDITION_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITION_MESSAGE}")
+  if(PRE_NEGATE)
+    if(${var})
+      if(PRE_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_MESSAGE}")
       else()
-        message(FATAL_ERROR "Error! Variable ${var} is true or not empty. The value of ${var} is ${${var}}.")
+        message(FATAL_ERROR
+          "Error! Variable ${var} is true or not empty. "
+          "The value of ${var} is ${${var}}.")
       endif()
     endif()
   else()
-    if (NOT ${var})
-      if (PRECONDITION_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITION_MESSAGE}")
+    if(NOT ${var})
+      if(PRE_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_MESSAGE}")
       else()
-        message(FATAL_ERROR "Error! Variable ${var} is false, empty or not set.")
+        message(FATAL_ERROR
+          "Error! Variable ${var} is false, empty or not set.")
       endif()
     endif()
   endif()
 endfunction()
 
+
 # Assert is 'NOT ${LHS} ${OP} ${RHS}' is true.
 function(precondition_binary_op OP LHS RHS)
-  cmake_parse_arguments(
-    PRECONDITIONBINOP # prefix
-    "NEGATE" # options
-    "MESSAGE" # single-value args
-    "" # multi-value args
+  set(options NEGATE)
+  set(single_value_args MESSAGE)
+  set(multi_value_args)
+
+  cmake_parse_arguments(PRE_BIN_OP
+    "${options}"
+    "${single_value_args}"
+    "${multi_value_args}"
     ${ARGN})
 
-  if (PRECONDITIONBINOP_NEGATE)
-    if (${LHS} ${OP} ${RHS})
-      if (PRECONDITIONBINOP_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITIONBINOP_MESSAGE}")
+  if(PRE_BIN_OP_NEGATE)
+    if(${LHS} ${OP} ${RHS})
+      if(PRE_BIN_OP_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_BIN_OP_MESSAGE}")
       else()
         message(FATAL_ERROR "Error! ${LHS} ${OP} ${RHS} is true!")
       endif()
     endif()
   else()
-    if (NOT ${LHS} ${OP} ${RHS})
-      if (PRECONDITIONBINOP_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITIONBINOP_MESSAGE}")
+    if(NOT ${LHS} ${OP} ${RHS})
+      if(PRE_BIN_OP_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_BIN_OP_MESSAGE}")
       else()
         message(FATAL_ERROR "Error! ${LHS} ${OP} ${RHS} is false!")
       endif()
     endif()
   endif()
 endfunction()
 
+
 # Translate a yes/no variable to the presence of a given string in a
 # variable.
 #
 # Usage:
-#   translate_flag(is_set flag_name var_name)
+#   translate_flag(is_set flag_name result_var)
 #
-# If is_set is true, sets ${var_name} to ${flag_name}. Otherwise,
-# unsets ${var_name}.
-function(translate_flag is_set flag_name var_name)
+# If is_set is true, sets ${result_var} to ${flag_name}. Otherwise,
+# unsets ${result_var}.
+function(translate_flag is_set flag_name result_var)
   if(${is_set})
-    set("${var_name}" "${flag_name}" PARENT_SCOPE)
+    set(${result_var} "${flag_name}" PARENT_SCOPE)
   else()
-    set("${var_name}" "" PARENT_SCOPE)
+    set(${result_var} "" PARENT_SCOPE)
```

Comment:
Fun fact, this is not "unsetting" in CMake. The doc-comment is just wrong here...

Reply:
Since you are making changes here:

1. I would consider renaming this to a better name that makes it clear what this is actually doing and or at least add an example to the doc-comment showing how this is used to rename variables in some of our cmake. The API either due to missing docs or a bad name (not sure which) is hard to remember functionally.  I, myself, always have to look it up to make sure I am not misusing it = (.

2. Can you use an early exit and flip the result around, i.e.:

```
if (NOT ${is_set}) {
    set(${result_var} "" PARENT_SCOPE)
    return()
}
set(${result_var} "${flag_name}" PARENT_SCOPE)
```